src/NuGet.Core/NuGet.LibraryModel/LibraryIdentity.cs

```diff
@@ -58,7 +58,7 @@ public override int GetHashCode()
 
             combiner.AddStringIgnoreCase(Name);
             combiner.AddObject(Version);
-            combiner.AddObject(Type);
```

Comment:
@benaadams Why not make `AddObject` into `AddObject<T>`? Instant win without other code changes.